infraestructure/handlers/song_controller.go

```diff
@@ -0,0 +1,74 @@
+package handlers
+
+import (
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/alexis-aguirre/golang-bootcamp-2020/domain/model"
+	"github.com/alexis-aguirre/golang-bootcamp-2020/registry"
+	"github.com/gorilla/mux"
+)
+
+//GetSongData retrieves the information of a song by it's name
+func GetSongData(w http.ResponseWriter, r *http.Request) {
+	variables := r.URL.Query()
+	log.Println(variables)
+	queryVariable, ok1 := variables["q"]
+	typeVariable, ok2 := variables["type"]
+
+	if !ok1 || !ok2 {
+		writeError(w, r, http.StatusBadRequest, "Malformed query")
+		return
+	}
+
+	if typeVariable[0] != "track" {
+		writeError(w, r, http.StatusNotImplemented, "Unsupported operation")
```

Comment:
```suggestion
		writeError(w, r, http.StatusNotImplemented, "Unsupported operation")
		return
```

I guess we also need to return after this error, right?